tests/js/server/dump/dump-move-shard.js

```diff
@@ -0,0 +1,126 @@
+//
+// @author Kaveh Vahedipour
+// //////////////////////////////////////////////////////////////////////////////
+
+var internal = require("internal");
+var jsunity = require("jsunity");
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function dumpTestSuite () {
+  'use strict';
+  var db = internal.db;
+
+  return {
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test the empty collection
+////////////////////////////////////////////////////////////////////////////////
+
+    testMoveShard : function () {
+
+      if (arango.getRole() !== "COORDINATOR") {
+        return;
+      }
+
+      let database = "UnitTestsDumpDst";
+      let collection = "UnitTestsDumpReplicationFactor1";
+      let planShards =
+          arango.GET("_admin/cluster/shardDistribution").results[collection].Plan;
+
+      /*eslint no-extend-native: ["error", { "exceptions": ["Array"] }]*/
+      Array.prototype.remove = function() {
+        var what, a = arguments, L = a.length, ax;
+          while (L && this.length) {
+          what = a[--L];
+          while ((ax = this.indexOf(what)) !== -1) {
+            this.splice(ax, 1);
+          }
+        }
+        return this;
+      };
+
+      var i = 0;
+      var pending = [];
+      Object.keys(planShards).forEach(
+        function (shard) {
+          let dbs = ["DBServer0001", "DBServer0002", "DBServer0003"];
+          let leader = planShards[shard].leader;
+          let follower = planShards[shard].followers[0];
+          dbs.remove(leader);
+          dbs.remove(follower);
+          let unused = dbs[0];
+          var toServer, fromServer;
+          let modulo = i % 4;
+
+          switch (modulo) {
+          case 0:
+            fromServer = leader; toServer = unused; break;
+          case 1:
+            fromServer = follower; toServer = unused; break;
+          case 2:
+            fromServer = leader; toServer = follower; break;
+          default:
+            return; break;
+          }
+
+          let body = {fromServer, toServer, database, collection, shard};
+          let result = arango.POST("_admin/cluster/moveShard", body);
+          assertTrue(!result.error);
+          assertTrue(result.code === 202);
+          pending.push(result.id);
+          i++;
+        });
+
+      while (pending.length > 0) { // wait for moveShard jobs to finish
+        var done = [];
+        pending.forEach(
+          function (jobId) {
+            let query = arango.GET("/_admin/cluster/queryAgencyJob?id=" + jobId);
+            if (query.status === "Finished") {
+              done.push(jobId);
+            }
+          }
+        );
+        done.forEach( function(jobId) { pending.remove(jobId); });
+        require("internal").sleep(0.25);
+      }
+
+      assertTrue(true);
```

Comment:
```suggestion
```

Reply:
What does the test actually test, except that dispatching moveShard jobs returns HTTP 202 and no error?
Is there another condition that we would like to see, e.g. that all shards _have_ been moved successfully as requested?